src/zabapgit_repo_impl.prog.abap

```diff
@@ -663,4 +690,37 @@ CLASS lcl_repo_srv IMPLEMENTATION.
 
   ENDMETHOD. "is_repo_installed
 
+  METHOD switch_repo_type.
+
+    DATA lo_repo TYPE REF TO lcl_repo.
+
+    FIELD-SYMBOLS <repo> LIKE LINE OF mt_list.
+
+    lo_repo = get( iv_key ).
+    READ TABLE mt_list ASSIGNING <repo> FROM lo_repo.
+    ASSERT sy-subrc IS INITIAL.
+    ASSERT iv_offline <> lo_repo->ms_data-offline.
+
+    IF iv_offline = abap_true. " On-line -> OFFline
+      lo_repo->set(
+        iv_url         = lcl_url=>name( lo_repo->ms_data-url )
+        iv_branch_name = ''
+        iv_sha1        = ''
+        iv_head_branch = ''
+        iv_offline     = abap_true
+      ).
+      CREATE OBJECT <repo> TYPE lcl_repo_offline
+        EXPORTING
+          is_data = lo_repo->ms_data.
+    ELSE. " OFFline -> On-line
+      lo_repo->set(
+        iv_offline     = abap_false
```

Comment:
not sure this will work, guess it will fail as the url from the offline repo is invalid? But I think its okay to merge, as it is an isolated feature